sourced/ml/models/quant.py

```diff
@@ -47,3 +47,11 @@ def dump(self):
         return """Schemes: %s""" % (
             sorted((v[0], "%d@%d" % (len(v[1]), len(next(iter(v[1].values()))) - 1))
                    for v in self.levels.items()))
+
+    def apply_quantization(self, extractors):
+        for extractor in extractors:
+            try:
+                extractor.quantize
```

Comment:
Did you forget `()`?

Reply:
no, the method `quantize` is only defined for the `children` extractor, so it raises this error for the other extractors even if it isn't called:

```
>>> from sourced.ml.extractors import GraphletBagExtractor
>>> GraphletBagExtractor().quantize
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'GraphletBagExtractor' object has no attribute 'quantize'
>>> from sourced.ml.extractors import ChildrenBagExtractor
>>> ChildrenBagExtractor().quantize
<bound method ChildrenBagExtractor.quantize of <sourced.ml.extractors.children.ChildrenBagExtractor object at 0x109f8de10>>
```

Reply:
I don't get why this function is needed still, it seems to have no effect

Reply:
It's pretty convoluted, but basically for all extractors with quantization (e.g. `children` only) it creates levels that are stored in `extractor.levels`. Then, when the features are extracted for all extractors you get to this in `children` (and I guess the same would go for other extractors):

```
        if not self.uast_to_bag.levels:
            # bypass NAMESPACE
            gen = self.uast_to_bag(uast).items()
        else:
            gen = super().extract(uast)
```
That is what this function does basically, just defines the level, either by computing them or loading from another model.